packages/fxa-settings/src/pages/ResetPassword/AccountRecoveryResetPassword/index.stories.tsx

```diff
@@ -3,41 +3,88 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 import React from 'react';
-import AccountRecoveryResetPassword, {
-  AccountRecoveryResetPasswordProps,
-} from '.';
-import AppLayout from '../../../components/AppLayout';
+import AccountRecoveryResetPassword, { AccountRecoveryResetPasswordProps } from '.';
 import { LocationProvider } from '@reach/router';
 import { Meta } from '@storybook/react';
-import { MOCK_ACCOUNT } from '../../../models/mocks';
+import { mockAppContext } from '../../../models/mocks';
 import { withLocalization } from '../../../../.storybook/decorators';
+import { AppContext, AppContextValue } from '../../../models';
+import { mockUrlSearchContext, mockUrlHashContext, mockStorageContext, mockAccount, mockRelierFactory, mockLocationContext } from './mocks';
+import { AuthUiErrors } from '../../../lib/auth-errors/auth-errors';
 
 export default {
   title: 'Pages/ResetPassword/AccountRecoveryResetPassword',
   component: AccountRecoveryResetPassword,
   decorators: [withLocalization],
 } as Meta;
 
-const storyWithProps = (props?: Partial<AccountRecoveryResetPasswordProps>) => {
-  const story = () => (
-    <LocationProvider>
-      <AppLayout>
-        <AccountRecoveryResetPassword
-          email={MOCK_ACCOUNT.primaryEmail.email}
-          linkStatus="valid"
-        />
-      </AppLayout>
-    </LocationProvider>
+const storyWithProps = (ctx:AppContextValue, props:AccountRecoveryResetPasswordProps) => {
+  return(
+    <AppContext.Provider value={ctx}>
+      <LocationProvider>
+        <AccountRecoveryResetPassword {...props} />
+      </LocationProvider>
+    </AppContext.Provider>
   );
-  return story;
 };
 
-export const WithValidLink = storyWithProps();
+function setup() {
+  const account = mockAccount();
+  const navigate = async (to:string|number, opts?:{}) => {
+    console.log('Would navigate to', to, opts);
+  }
+  const urlSearchContext = mockUrlSearchContext();
```

Comment:
@LZoog Mocking UrlSearchContext without having to mess with window.location.search. This is pretty clean IMO, and solves a couple problems I ran into with query params and storybook.